src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffoldSteps/ValidateBlazorIdentityStep.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.DotNet.Scaffolding.Core.Scaffolders;
+using Microsoft.DotNet.Scaffolding.Core.Steps;
+using Microsoft.DotNet.Scaffolding.Internal.Services;
+using Microsoft.DotNet.Scaffolding.TextTemplating.DbContext;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Common;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Helpers;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Models;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.ScaffoldSteps.Settings;
+using Microsoft.Extensions.Logging;
+using static Microsoft.DotNet.Scaffolding.Internal.Constants;
+
+namespace Microsoft.DotNet.Tools.Scaffold.AspNet.ScaffoldSteps;
+
+internal class ValidateBlazorIdentityStep : ScaffoldStep
+{
+    private readonly IFileSystem _fileSystem;
+    private readonly ILogger _logger;
+    public bool Overwrite { get; set; }
+    public string? Project { get; set; }
+    public bool Prerelease { get; set; }
+    public string? DatabaseProvider { get; set; }
+    public string? DataContext { get; set; }
+    public ValidateBlazorIdentityStep(
+        IFileSystem fileSystem,
+        ILogger<ValidateBlazorIdentityStep> logger)
+    {
+        _fileSystem = fileSystem;
+        _logger = logger;
+    }
+
+    public override async Task<bool> ExecuteAsync(ScaffolderContext context, CancellationToken cancellationToken = default)
```

Comment:
There's _some_ validation in here (in the aptly named ValidateBlazorIdentitySettings call on the first line in particular), but really most of it seems like setup (filling in codeModifierProperties, filling in dbContextProperties, adding everything to context.Properties, etc. 

Are all of these things that are used by multiple steps afterwards, thus it makes sense to have one step that pre-calculates things? I guess having that in a Validation step makes sense... though I wonder if we should have a better (more "intentional") mechanism for that. Kind of the same way we were thinking about with validation overall.

Reply:
yep I would say majority of the validation of these scaffolders is the same so we should be able to pull that out and have scaffolder specific validation after. Adding a TODO for now